vantage6/server/resource/role.py

```diff
@@ -200,21 +200,24 @@ def post(self):
         if denied:
             return denied, HTTPStatus.UNAUTHORIZED
 
-        # if trying to create a role for another organization
-        if data["organization_id"] and not self.r.c_glo.can():
-            return {'msg': 'You cannot create roles for other organizations'},\
-                HTTPStatus.UNAUTHORIZED
-        elif data["organization_id"] and self.r.c_glo.can():
-            organization_id = data["organization_id"]
-
-            # verify that the organization exists
-            if not db.Organization.get(organization_id):
-                return {'msg': f'organization "{organization_id}" does not '
-                        'exist!'}, HTTPStatus.NOT_FOUND
-        elif (not data['organization_id'] and self.r.c_glo.can()) or \
-                self.r.c_org.can():
-            organization_id = g.user.organization_id
-        else:
+        # set the organization id
+        organization_id = (
+            data['organization_id']
+            if data['organization_id'] else g.user.organization_id
+        )
+        # verify that the organization for which we creat a role exists
+        if (organization_id != g.user.organization_id and
```

Comment:
Did you do this for performance? I do not completely gey why we need to check that the `organization_id != g.user.organization_id`. The organization needs to exist in either case?

Reply:
You're right, I think it's clearer without the extra check. Performance might indeed be somewhat worse but I don't see it being a large effect. So I removed the check.